src/app/Main.java

```diff
@@ -56,27 +60,52 @@ public static void main(String[] args) {
                     String userNameReservation = sn.next();
                     System.out.println("type lastname of user");
                     String userLastnameReservation = sn.next();
-                    boolean usuarioEncontrado = userService.userExists(userNameReservation, userLastnameReservation);
+                    boolean userFound = userService.userExists(userNameReservation, userLastnameReservation);
 
                     System.out.println("type name of book");
                     String nameBookReservation = sn.next();
                     System.out.println("type name of author");
                     String nameAuthorBookReservation = sn.next();
                     System.out.println("type year of book");
                     int yearBookReservation = sn.nextInt();
-                    boolean libroEncontrado = bookService.bookExists(nameBookReservation, nameAuthorBookReservation, yearBookReservation);
+                    boolean bookFound = bookService.bookExists(nameBookReservation, nameAuthorBookReservation, yearBookReservation);
 
-                    if (usuarioEncontrado && !libroEncontrado) {
+                    if (userFound && bookFound) {
```

Comment:
This logic is part of the business logic of your application, try to move everything to the layer service.

The business rule leading here will be:
"Only existing users can create reservations over existing books"

SInce that is a business rule, makes sense that it lives on the service layer, not on the control layer (the menu)